core/spec/models/spree/stock/allocator/base_spec.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+module Spree
+  module Stock
+    module Allocator
+      RSpec.describe Base, type: :model do
+        subject { described_class.new(desired, availability) }
+
+        let(:desired) { double(Spree::StockQuantities) }
+        let(:availability) { double(Spree::Stock::Availability) }
+
+        before do
+          allow(availability).to receive(:on_hand_by_stock_location_id)
+            .and_return(double(Spree::StockQuantities))
+          allow(availability).to receive(:backorderable_by_stock_location_id)
+            .and_return(double(Spree::StockQuantities))
+        end
+
+        describe '#allocate_on_hand' do
+          it 'calls allocate_inventory method' do
+            is_expected.to receive(:allocate_inventory).with(availability.on_hand_by_stock_location_id)
```

Comment:
This (and the test above) has a couple of different issues:

- You shouldn't test private methods, just the public interface.
- You shouldn't expect that a method will be called on the test subject, but rather test the side effects.